File: 백준/Silver/1929. 소수 구하기/소수 구하기.py
import math
a,b = map(int,input().split())

arr = [0]*(b+1)

for i in range(2,b+1):
    arr[i] = i


for i in range(2, int(math.sqrt(b))+1):
    if arr[i]==0:
        continue
    for j in range(i+i,b+1,i):
        arr[j]=0


for i in range(a,b+1):
    if arr[i] !=0:
        print(arr[i])

File: 백준/Silver/1929. 소수 구하기/소수 구하기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());
		int n = Integer.parseInt(st.nextToken());
		int m = Integer.parseInt(st.nextToken());
		int [] arr = new int [m+1];
		
		for(int i=2;i<=m;i++) {
			arr[i] = i;
		}
		
		for(int i=2;i<=Math.sqrt(m);i++) {
			if(arr[i]==0) {
				continue;
			}
			for(int j = i+i;j<=m;j=j+i) {
				arr[j] = 0;
			}
		}
		for(int i=n;i<=m;i++) {
			if(arr[i] !=0) {
				System.out.println(arr[i]);
			}
		}
		
	}

}
